src/python/turicreate/test/test_sframe_s3.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2020 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+#
+from __future__ import print_function as _  # noqa
+from __future__ import division as _  # noqa
+from __future__ import absolute_import as _  # noqa
+from ..data_structures.sframe import SFrame
+
+import tempfile
+import os
+import math
+import shutil
+import pytest
+import boto3
+
+
+@pytest.mark.skipif(
+    os.environ.get("DTURI_ENABLE_SF_S3") is None,
+    reason="slow IO test; enabled when needed",
+)
+class TestSFrameS3(object):
+    @classmethod
+    def setup_class(self):
+        self.my_tempdir = tempfile.mkdtemp()
+        self.s3_client = boto3.client(
+            "s3",
+            endpoint_url=os.environ["TURI_S3_ENDPOINT"],
+            region_name=os.environ["TURI_S3_REGION"],
+        )
+        self.bucket = "tc_qa"
+        self.s3_prefix = "integration/manual/"
+
+    @classmethod
+    def teardown_class(self):
+        shutil.rmtree(self.my_tempdir)
+
+    def _assert_sarray_equal(self, sa1, sa2):
+        l1 = list(sa1)
+        l2 = list(sa2)
+        self.assertEqual(len(l1), len(l2))
+        for i in range(len(l1)):
+            v1 = l1[i]
+            v2 = l2[i]
+            if v1 is None:
+                self.assertEqual(v2, None)
+            else:
+                if type(v1) == dict:
+                    self.assertEqual(len(v1), len(v2))
+                    for key in v1:
+                        self.assertTrue(key in v1)
+                        self.assertEqual(v1[key], v2[key])
+
+                elif hasattr(v1, "__iter__"):
+                    self.assertEqual(len(v1), len(v2))
+                    for j in range(len(v1)):
+                        t1 = v1[j]
+                        t2 = v2[j]
+                        if type(t1) == float:
+                            if math.isnan(t1):
+                                self.assertTrue(math.isnan(t2))
+                            else:
+                                self.assertEqual(t1, t2)
+                        else:
+                            self.assertEqual(t1, t2)
+                else:
+                    self.assertEqual(v1, v2)
+
+    def _test_sframe_equal(self, sf1, sf2):
```

Comment:
Can you just use the [method we already have for doing this](https://github.com/apple/turicreate/blob/f57ca7c6af95c7497b01ca3326e945f4d5057a1c/src/python/turicreate/util/__init__.py#L276)?

Reply:
Yeah, why not. I hope I know it earlier.